src/cregistry/sql.c

```diff
@@ -224,6 +224,16 @@ int create_tables(sqlite3* db, reg_error* errPtr) {
             " ON DELETE CASCADE"
             ")",
 
+        /* Files registered to ports in snapshots.
+           Needed to resolve file-based dependencies. */
+        "CREATE TABLE registry.snapshot_files ("
+              "id INTEGER"
+            ", path TEXT"
+            ", FOREIGN KEY(id) REFERENCES snapshot_ports(id)"
+            " ON DELETE CASCADE"
```

Comment:
I realize the code above already uses `FOREIGN KEY` and `ON DELETE CASCADE`, but unfortunately I don't think that currently works. https://www.sqlite.org/foreignkeys.html says this needs sqlite >= 3.6.19 (which I don't think is a huge issue), but it also says in [section 2](https://www.sqlite.org/foreignkeys.html#fk_enable) that foreign key support must be explicitly enabled using
```
PRAGMA foreign_keys = ON;
```
for each connection.

It seems that without this, the `ON DELETE CASCADE` just does nothing:

```
$ sqlite3 no-fk.db
-- Loading resources from /Users/clemens/.sqliterc
.headers ON
.separator "\t"
.nullvalue "NULL"
.mode box
SQLite version 3.46.0 2024-05-23 13:25:27
Enter ".help" for usage hints.
sqlite> CREATE TABLE ports (id INTEGER PRIMARY KEY, name TEXT COLLATE NOCASE);
CREATE TABLE ports (id INTEGER PRIMARY KEY, name TEXT COLLATE NOCASE);
sqlite> CREATE TABLE files (id INTEGER, path TEXT, FOREIGN KEY(id) REFERENCES ports(id) ON DELETE CASCADE);
CREATE TABLE files (id INTEGER, path TEXT, FOREIGN KEY(id) REFERENCES ports(id) ON DELETE CASCADE);
sqlite> INSERT INTO ports (id, name) VALUES (1, 'port1'), (2, 'port2');
INSERT INTO ports (id, name) VALUES (1, 'port1'), (2, 'port2');
sqlite> INSERT INTO files (id, path) VALUES (1, '/path/port1/foo'), (1, '/path/port1/bar'), (2, '/path/port2/foo'), (2, '/path/port2/bar');
INSERT INTO files (id, path) VALUES (1, '/path/port1/foo'), (1, '/path/port1/bar'), (2, '/path/port2/foo'), (2, '/path/port2/bar');
sqlite> DELETE FROM ports WHERE id = 1;
DELETE FROM ports WHERE id = 1;
sqlite> SELECT * FROM files WHERE id = 1;
SELECT * FROM files WHERE id = 1;
┌────┬─────────────────┐
│ id │      path       │
├────┼─────────────────┤
│ 1  │ /path/port1/foo │
│ 1  │ /path/port1/bar │
└────┴─────────────────┘
sqlite> ^D
```

It does work with it enabled:
```
$ sqlite3 fk.db
-- Loading resources from /Users/clemens/.sqliterc
.headers ON
.separator "\t"
.nullvalue "NULL"
.mode box
SQLite version 3.46.0 2024-05-23 13:25:27
Enter ".help" for usage hints.
sqlite> PRAGMA foreign_keys = ON;
PRAGMA foreign_keys = ON;
sqlite> CREATE TABLE ports (id INTEGER PRIMARY KEY, name TEXT COLLATE NOCASE);
CREATE TABLE ports (id INTEGER PRIMARY KEY, name TEXT COLLATE NOCASE);
sqlite> CREATE TABLE files (id INTEGER, path TEXT, FOREIGN KEY(id) REFERENCES ports(id) ON DELETE CASCADE);
CREATE TABLE files (id INTEGER, path TEXT, FOREIGN KEY(id) REFERENCES ports(id) ON DELETE CASCADE);
sqlite> INSERT INTO ports (id, name) VALUES (1, 'port1'), (2, 'port2');
INSERT INTO ports (id, name) VALUES (1, 'port1'), (2, 'port2');
sqlite> INSERT INTO files (id, path) VALUES (1, '/path/port1/foo'), (1, '/path/port1/bar'), (2, '/path/port2/foo'), (2, '/path/port2/bar');
INSERT INTO files (id, path) VALUES (1, '/path/port1/foo'), (1, '/path/port1/bar'), (2, '/path/port2/foo'), (2, '/path/port2/bar');
sqlite> DELETE FROM ports WHERE id = 1;
DELETE FROM ports WHERE id = 1;
sqlite> SELECT * FROM files WHERE id = 1;
SELECT * FROM files WHERE id = 1;
sqlite> ^D
```

Reply:
I guess I would have discovered this when trying to implement snapshot deletion, which is still on the todo list. :)